File: Data/Models/CashierDetail.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nebula.Data.Models
{
    public enum TypeOperation
    {
        CajaChica,
        Comprobante
    }

    /// <summary>
    /// Detalle de caja diaria.
    /// </summary>
    public class CashierDetail
    {
        public int Id { get; set; }

        /// <summary>
        /// Clave foránea CajaDiaria.
        /// </summary>
        public int? CajaDiariaId { get; set; }

        /// <summary>
        /// Propiedad de Relación con Caja diaria.
        /// </summary>
        [ForeignKey("CajaDiariaId")]
        public CajaDiaria CajaDiaria { get; set; }

        /// <summary>
        /// Clave foránea Factura.
        /// </summary>
        public int? InvoiceId { get; set; }

        /// <summary>
        /// Propiedad de Relación con Factura.
        /// </summary>
        [ForeignKey("InvoiceId")]
        public Invoice Invoice { get; set; }

        /// <summary>
        /// Tipo de Operación.
        /// </summary>
        public TypeOperation? TypeOperation { get; set; }

        /// <summary>
        /// Fecha de registro.
        /// </summary>
        [DataType((DataType.Date))]
        public DateTime StartDate { get; set; }

        /// <summary>
        /// Serie y Número de documento.
        /// </summary>
        [MaxLength(250)]
        public string Document { get; set; }

        /// <summary>
        /// Nombre Contacto.
        /// </summary>
        [MaxLength(250)]
        public string Contact { get; set; }

        /// <summary>
        /// Observación de la Operación.
        /// </summary>
        [MaxLength(250)]
        public string Glosa { get; set; }

        /// <summary>
        /// Medios de Pago.
        /// </summary>
        public int? PaymentMethod { get; set; }

        /// <summary>
        /// Movimiento de efectivo,
        /// (Entrada|Salida).
        /// </summary>
        [MaxLength(250)]
        public string Type { get; set; }

        /// <summary>
        /// Monto de la Operación.
        /// </summary>
        public decimal? Total { get; set; }
    }
}

File: Angular/src/app/cashier/interfaces/cashier-detail.ts
export class CashierDetail {
  constructor(
    public id: number | any = null, // identificador clave primaria.
    public cajaDiariaId: number | any = null, // identificador de caja diaria.
    public invoiceId: number | any = null, // identificador del comprobante de venta.
    public typeOperation: number = 0, // tipo operación: <Caja Chica/Comprobante>
    public startDate: any = null, // fecha de registro.
    public document: string = '', // serie y número del comprobante.
    public contact: string = '', // nombre de contacto.
    public glosa: string = '', // observación de la operación.
    public paymentMethod: number = 0, // medios de pago: <dinero en efectivo, depósito en cuenta, billetera digital>
    public type: string = 'ENTRADA', // tipo de operación: <ENTRADA/SALIDA>
    public total: number = 0,) {
  }
}
